docs/rfc/improving-sync.md

```diff
@@ -0,0 +1,179 @@
+# Store RFC: Improving Sync
+
+## Metadata
+
+- **Authors**: [Matt Ramotar](https://github.com/matt-ramotar)
+- **Reviewers**: _TBD_
+- **Started**: February 22, 2025
+
+## Introduction
+
+Store is our solution for working with data in Kotlin. It orchestrates CRUD operations with offline-first capabilities. It simplifies data flow between local and remote sources, ensuring that applications can remain responsive even under unreliable network conditions.
+
+Store5 specifically introduced the `MutableStore` API to unify local and remote data synchronization. It provides mechanisms for create, update, and delete operations. However, we’ve identified critical gaps around resilience (automated retry, robust background sync) and deletion handling (both local and server-initiated removals). Along with these, we also need customizable conflict resolution strategies to better address real-world complexities.
+
+The overarching goal of this RFC is to enhance Store5’s synchronization features without introducing breaking changes. In particular:
```

Comment:
This implies there would be some changes proposed if we were interested in breaking changes, that aren't called out here. Is there anything else in the sync category, or something we would have proposed differently here, if that was on the table and we'd want to discuss those with a major version bump at some point in the future?